src/main/java/de/tum/in/www1/artemis/service/FileService.java

```diff
@@ -799,4 +817,34 @@ public Path writeObjectToJsonFile(Object object, ObjectMapper objectMapper, Path
         }
         return path;
     }
+
+    /**
+     * Merge the PDF files located in the given paths.
+     *
+     * @param paths list of paths to merge
+     * @return byte array of the merged file
+     */
+    public byte[] mergePdfFiles(List<String> paths) {
```

Comment:
In the `FileResource` you check if this was successful by testing the array for `null` and emptiness. I think it would make sense to either return an `Optional<byte[]>` or throw an exception to tell the caller that this operation can fail just by looking at the type of the function. This in turn would then also force them to handle the failure case.